orchagent/portsorch.cpp

```diff
@@ -3077,10 +3172,19 @@ bool PortsOrch::addVlanMember(Port &vlan, Port &port, string &tagging_mode)
     }
 
     /* a physical port may join multiple vlans */
-    VlanMemberEntry vme = {vlan_member_id, sai_tagging_mode};
+    VlanMemberEntry vme = {vlan.m_alias, vlan_member_id, sai_tagging_mode};
     port.m_vlan_members[vlan.m_vlan_info.vlan_id] = vme;
     m_portList[port.m_alias] = port;
     vlan.m_members.insert(port.m_alias);
+    if (port.m_oper_status == SAI_PORT_OPER_STATUS_UP)
+    {
+        auto old_count = vlan.m_up_member_count;
+        vlan.m_up_member_count++;
+        if (old_count == 0)
+        {
+        	updateDbVlanOperStatus(vlan, "up");
```

Comment:
Link kernel should manage the vlan operations status, why do we want to manage ourselves?  Also, we do want VLAN up even all ports are down in some cases (by default for some users).

Reply:
If vlan netdevice is up by default, it has undesired effects. so we make the vlan up only if there is a member which is oper up. portsorch sends vlan oper state to vlanmgr and vlanmgr sets the vlan admin state in kernel accordingly (void VlanMgr::doTask(NotificationConsumer &consumer)).
If it's needed to set vlan state UP for any other reason, it can be decided in orchagent and communicated to vlanmgr accordingly.